services-camel/src/test/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesRouterTest.java

```diff
@@ -0,0 +1,220 @@
+        template.sendBodyAndHeaders("", createEvent("image/png", Cdr.FileObject.getURI()));
+        verify(binaryInfoProcessor).process(any(Exchange.class));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void routeNonBinary() throws Exception {
+        getMockEndpoint("mock:direct:image.derivatives.destroy").expectedMessageCount(0);
+        getMockEndpoint("mock:direct:fulltext.derivatives.destroy").expectedMessageCount(0);
+
+        createContext(DESTROY_DERIVATIVES_ROUTE);
+        template.sendBodyAndHeaders("", createEvent("application", Cdr.FileObject.getURI()));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    @Test
+    public void destroyTextDerivative() throws Exception {
+        createContext(DESTROY_FULLTEXT_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("text/plain", Cdr.FileObject.getURI()));
+
+        verify(destroyFulltextProcessor).process(any(Exchange.class));
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyImageThumbnailDerivative() throws Exception {
+        createContext(DESTROY_IMAGE_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("image/png", Cdr.FileObject.getURI()));
+
+        verify(destroySmallThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyImageThumbnailDerivativeCollection() throws Exception {
+        createContext(DESTROY_IMAGE_ROUTE);
+
+        template.sendBodyAndHeaders("", createEvent("image/*", Cdr.Collection.getURI()));
+
+        verify(destroySmallThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyImageSrcCopyCollection() throws Exception {
+        createContext(DESTROY_SRC_COPY);
+
+        Map<String, Object> headers = createEvent("image/*", Cdr.Collection.getURI());
+        headers.put("CollectionThumb", true);
+        template.sendBodyAndHeaders("", headers);
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
```

Comment:
since you're directly testing the `CdrDestroyCollectionUpload` route here, it would be fine to only verify `destroyCollectionSrcImgProcessor` is called. The other processors aren't related to this route, and it confused me initially that the thumbnail processors weren't called for cleanup of a collection thumbnail.